src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealm.java

```diff
@@ -574,6 +592,12 @@ private void writeIdentity(final XMLStreamWriter streamWriter, final LoadedIdent
             streamWriter.writeEndDocument();
         }
 
+        public void dispose() {
+            // Release the write lock for this realm identity
+            final ReentrantReadWriteLock realmIdentityLock = getRealmIdentityLockForName(name);
+            realmIdentityLock.writeLock().unlock();
```

Comment:
This will fail if/when the dispose or subsequent authentication mechanism methods are called from different threads because the standard Lock implementations are owned by Threads.